frontend/pkg/frontend/frontend_test.go

```diff
@@ -219,12 +238,39 @@ func TestSubscriptionsPUT(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			if test.subDoc != nil {
-				err := f.dbClient.CreateSubscriptionDoc(context.TODO(), test.subDoc)
-				if err != nil {
-					t.Fatal(err)
+			// MiddlewareLockSubscription
+			// (except when MiddlewareValidateStatic fails)
+			mockDBClient.EXPECT().
+				GetLockClient().
+				MaxTimes(1)
+			if test.expectedStatusCode != http.StatusBadRequest {
+				// ArmSubscriptionPut
+				mockDBClient.EXPECT().
+					GetSubscriptionDoc(gomock.Any(), gomock.Any()).
+					DoAndReturn(func(context.Context, string) (*database.SubscriptionDocument, error) {
```

Comment:
nit: This is the second time I see this in-line function - should we consider making it a named function defined in this test suite?

Reply:
It's a little tricky due to the variable scoping of `test.subDoc`.  But I suppose a named function could take a document pointer argument and return the inline function.  Bonus points if I can make it a generic function for any document type.

Reply:
Turned out simpler than I thought, and I did manage to make it a generic function.